### ISSUE TITLE
Overload argument expansion incorrectly skipped when call includes positional arguments

### ISSUE THREAD

**Issue by carljm** [astral-sh/ty#4258]
### Summary

An overloaded call with a union-typed keyword argument incorrectly produces
`no-matching-overload` when the call also contains an unpacked positional
argument, including an empty tuple.

### Reproduction

```python
from typing import overload

@overload
def read(*ids: str, kind: int) -> int: ...

@overload
def read(*ids: str, kind: None) -> str: ...

def read(*ids: str, kind: int | None) -> int | str:
    return 0

def example(ids: tuple[str], kind: int | None) -> None:
    read("a", kind=kind)   # Accepted.
    read(*(), kind=kind)   # no-matching-overload
    read(*ids, kind=kind)  # no-matching-overload
```

### Expected behavior

All three calls should type-check and have return type `int | str`.

The positional arguments are valid for both overloads: `*()` supplies zero
arguments, and `*ids` supplies one `str`. Expanding `kind: int | None` should
match the first overload for `int` and the second overload for `None`.

### Actual behavior

Both calls containing unpacked positional arguments produce
`no-matching-overload`.

### Possible cause

The optimization before overload argument expansion checks whether each
non-expandable argument is assignable to a corresponding overload parameter.
For `*ids`, it compares `tuple[str]` against the variadic parameter’s element
type `str`; for `*()`, there are no matched parameters. Both cases incorrectly
cause ty to skip expansion of `kind`.

Running with `TY_LOG=ty_python_semantic::types::call::bind=debug` confirms:

```text
Argument at 0 (`tuple[()]`) is not assignable to any of the remaining overloads, skipping argument type expansion
Argument at 0 (`tuple[str]`) is not assignable to any of the remaining overloads, skipping argument type expansion
```

Related historical context: #868.
